### PR TITLE
Add a typevar with a default to `io.TextIOWrapper`

### PR DESCRIPTION
**Pull request by sobolevn**
Let's try this :)

Refs https://github.com/python/typeshed/pull/11420
Refs https://github.com/python/typeshed/issues/11418

### CODE REVIEW

File: stdlib/io.pyi [hunk 1]
```diff
@@ -173,12 +173,12 @@ class _WrappedBuffer(Protocol):
     # def seek(self, offset: Literal[0], whence: Literal[2]) -> int: ...
     # def tell(self) -> int: ...
 
-# TODO: Should be generic over the buffer type, but needs to wait for
-# TypeVar defaults.
-class TextIOWrapper(TextIOBase, TextIO):  # type: ignore[misc]  # incompatible definitions of write in the base classes
+_BufferT = TypeVar("_BufferT", bound=_WrappedBuffer, default=_WrappedBuffer)
```
Comment (AlexWaygood) [inline stdlib/io.pyi:176]: Should the TypeVar be covariant?

```suggestion
_BufferT = TypeVar("_BufferT", bound=_WrappedBuffer, default=_WrappedBuffer, covariant=True)
```

Comment (sobolevn) [reply stdlib/io.pyi]: Let's try it.

**Review by srittau — APPROVED** [review]
LGTM, the new primer hit look like risky code that should at least get a `# type: ignore`.

**Comment by sobolevn** [comment]
Yeah, I agree - this is a risky one. Revert is always an option. IO is hard :)

**Pull request merged**